src/polycubed/src/cubes_dump.cpp

```diff
@@ -0,0 +1,336 @@
+  kill_ = true;
+  waitForUpdate_.notify_one();
+  save_in_file_thread_->join();
+}
+
+void CubesDump::Enable() {
+  enabled_ = true;
+}
+
+void CubesDump::UpdateCubesConfig(const std::string& resource,
+                                  const nlohmann::json& body,
+                                  const ListKeyValues &keys,
+                                  Rest::Resources::Endpoint::Operation opType,
+                                  Rest::Resources::Endpoint::ResourceType resType) {
+
+  // TODO: remove this check
+  if (configuration::config.getCubesNoDump()) {
+    return;
+  }
+
+  std::string resourcePath = resource.substr(RestServer::base.size(), resource.size());
+
+  std::stringstream ssResource(resourcePath);
+  std::vector<std::string> resItem;
+  std::string tokenResource;
+
+  while (std::getline(ssResource, tokenResource, '/')) { /* Get path elements */
+    resItem.push_back(tokenResource);
+  }
+
+  /*
+   * Depending on the operation, we look into the resource string, the body
+   * and the ListKeyValues to update the configuration of the cubes in the map
+   * (<cubeName, cubeConfiguration>) we have in memory.
+   * If it is not the initial topology load, the whole configuration is saved
+   * to file after each modification by a separate thread, to avoid to overload
+   * the server thread.
+   */
+  std::lock_guard<std::mutex> guardConfigMutex(cubesConfigMutex_);
+  switch (opType) {
+  case Operation::kCreate:
+  case Operation::kReplace:
+    UpdateCubesConfigCreateReplace(resItem, body, keys, resType);
+    break;
+
+  case Operation::kUpdate:
+    UpdateCubesConfigUpdate(resItem, body, keys, resType);
+    break;
+
+  case Operation::kDelete:
+    UpdateCubesConfigDelete(resItem, body, keys, resType);
+    break;
+  }
+
+  // if startup is true, then it is the first load and the save thread must not be notified
+  if (enabled_) {
+    toSave_++;
+    waitForUpdate_.notify_one();
+  }
+}
+
+void CubesDump::UpdatePortPeer(const std::string& cubeName, const std::string& cubePort, std::string peer) {
+
+  // TODO: remove this check
+  if (configuration::config.getCubesNoDump()) {
+    return;
+  }
+
+  std::lock_guard<std::mutex> guardConfigMutex(cubesConfigMutex_);
+  auto *cubePortsArray = &(cubesConfig_.at(cubeName).at("ports"));
+  auto *portToConnect = &*(std::find_if(cubePortsArray->begin(), cubePortsArray->end(),
+          [cubePort](auto const& elem){return elem.at("name") == cubePort;}));
+  if (peer.empty()) {
+    portToConnect->erase("peer");
+  } else {
+    nlohmann::json jsonPeer = nlohmann::json::object();
+    jsonPeer["peer"] = peer;
+    jsonPeer.update(*portToConnect);
+    portToConnect->clear();
+    portToConnect->update(jsonPeer);
+  }
+
+  // if startup is true, then it is the first load and the save thread must not be notified
+  if (enabled_) {
+    toSave_++;
+    waitForUpdate_.notify_one();
+  }
+}
+
+void CubesDump::UpdateCubesConfigCreateReplace(const std::vector<std::string> &resItem,
```

Comment:
Add a comment: what is this function about?
In general, functions should have a short comment at the beginning, explaining what they are doing.
Please extend the comment to the other functions as well.